### DISCUSSION TITLE
Custom Runtimes

### DISCUSSION THREAD

**Discussion by theWebKeyGuy** [hammerstonedev/sidecar#120] [Q&A]
Hello, just wanted to make a recommendation. It would be awesome if we could build custom runtimes so that way we can build a lmabda function let's say for PHP instead of using node.js or python. On the other hand, what I want to suggest is also maybe building a runtime for PHP.

**Comment by adrianspacely** [comment]
[hammerstonedev/sidecar-php](https://github.com/hammerstonedev/sidecar-php) 👀

**Comment by ppavlovdev** [comment]
@theWebKeyGuy 

This project allows you to define custom runtime in your Lambda definition class:
https://github.com/hammerstonedev/sidecar/blob/edbc0ad6be1b1de37ec52efbebe37724c33d0250/src/Runtime.php#L43

Then just add `bootstrap` runtime handler in your packages and provide required binaries/libraries using Lambda Layer (preferred way).

UPD: AWS detailed tutorial - [PHP Custom Runtime](https://aws.amazon.com/blogs/apn/aws-lambda-custom-runtime-for-php-a-practical-example/)
